components/button-group/button-group.js

```diff
@@ -0,0 +1,162 @@
+import '../dropdown/dropdown-menu.js';
+import '../button/button.js';
+import '../button/button-subtle.js';
+import '../dropdown/dropdown.js';
+import '../dropdown/dropdown-button.js';
+import '../dropdown/dropdown-button-subtle.js';
+import '../dropdown/dropdown-more.js';
+import '../menu/menu.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { DynamicResizeMixin } from '../../mixins/dynamic-resize-mixin.js';
+import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
+import { offscreenStyles } from '../offscreen/offscreen.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+/**
+ *
+ * A button group component that can be used to display a set of buttons
+ *
+ * @slot - Buttons, dropdown buttons, links or other items to be added to the container
+ */
+class ButtonGroup extends RtlMixin(LocalizeCoreElement(DynamicResizeMixin(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Setting this property to true will change the look of the dopdown button to a subtle button for subtle button groups
+			 */
+			subtle: {
+				type: Boolean,
+				attribute: 'subtle',
+			},
+
+		};
+	}
+
+	static get styles() {
+		return [offscreenStyles,
+			css`
+			:host {
+				display: block;
+
+			}
+			
+			:host([subtle]) {
+				--d2l-button-icon-fill-color: var(--d2l-color-celestine);
+				--d2l-button-icon-fill-color-hover: var(--d2l-color-celestine-minus-1);
+			}
+
+			:host .d2l-button-group-container {
+				display: flex;
+				flex: 0 1 auto;
+				flex-wrap: wrap;
+			}
+
+			:host .d2l-button-group-container ::slotted(d2l-button),
+			:host .d2l-button-group-container ::slotted(d2l-button-icon),
+			:host .d2l-button-group-container ::slotted(d2l-link),
+			:host .d2l-button-group-container ::slotted(span),
+			:host .d2l-button-group-container ::slotted(d2l-dropdown:not(.d2l-overflow-dropdown)),
+			:host .d2l-button-group-container ::slotted(d2l-dropdown-button),
+			:host .d2l-button-group-container ::slotted(d2l-dropdown-button-subtle),
+			:host .d2l-button-group-container ::slotted(.d2l-button-group-custom-item) {
+				margin-right: 0.6rem;
+			}
+
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-button),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-button-icon),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-link),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(span),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-dropdown:not(.d2l-overflow-dropdown)),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-dropdown-button),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-dropdown-button-subtle),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(.d2l-button-group-custom-item) {
+				margin-left: 0.6rem;
+				margin-right: 0;
+			}
+
+			:host([subtle]) .d2l-button-group-container ::slotted(d2l-button-subtle) {
+				margin-right: 0.2rem;
```

Comment:
Interesting... subtle buttons have a different margin?

Reply:
Yeah, since there is no background color/borders around the subtle buttons they look really spread out with the `0.6rem` margins. This was suggested by Jeff so that the space between the button text looks similar to the non subtle buttons